gradle-baseline-java/src/main/groovy/com/palantir/baseline/tasks/ClassUniquenessAnalyzer.java

```diff
@@ -0,0 +1,155 @@
+/*
+ * (c) Copyright 2018 Palantir Technologies Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.baseline.tasks;
+
+import static java.util.stream.Collectors.toSet;
+
+import com.google.common.hash.HashCode;
+import com.google.common.hash.Hashing;
+import com.google.common.hash.HashingInputStream;
+import com.google.common.io.ByteStreams;
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.IOException;
+import java.time.Duration;
+import java.time.Instant;
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Set;
+import java.util.jar.JarEntry;
+import java.util.jar.JarInputStream;
+import java.util.stream.Collectors;
+import org.gradle.api.artifacts.Configuration;
+import org.gradle.api.artifacts.ModuleVersionIdentifier;
+import org.gradle.api.artifacts.ResolvedArtifact;
+import org.slf4j.Logger;
+
+public final class ClassUniquenessAnalyzer {
+
+    private final Map<Set<ModuleVersionIdentifier>, Set<String>> jarsToClasses = new HashMap<>();
+    private final Map<String, Set<HashCode>> classToHashCodes = new HashMap<>();
+    private final Logger log;
+
+    public ClassUniquenessAnalyzer(Logger log) {
+        this.log = log;
+    }
+
+    public void analyzeConfiguration(Configuration configuration) {
+        Instant before = Instant.now();
+        Set<ResolvedArtifact> dependencies = configuration
+                .getResolvedConfiguration()
+                .getResolvedArtifacts();
+
+        // we use these temporary maps to accumulate information as we process each jar,
+        // so they may include singletons which we filter out later
+        Map<String, Set<ModuleVersionIdentifier>> classToJars = new HashMap<>();
+        Map<String, Set<HashCode>> tempClassToHashCodes = new HashMap<>();
+
+        dependencies.stream().forEach(resolvedArtifact -> {
+            File file = resolvedArtifact.getFile();
+            if (!file.exists()) {
```

Comment:
probably wanna filter out non-jars too to avoid spurious errors in line 92

Reply:
TBH I'd kinda like to just get an MVP released and see get a few repos to pick it up and see what we encounter!  The `-sources.jar` and `-javadoc.jar` thing doesn't seem to cause a problem because we only consider files ending in `.class`.